File: 392.is-subsequence.py
#
# @lc app=leetcode id=392 lang=python
#
# [392] Is Subsequence
#

# @lc code=start
class Solution(object):
    def isSubsequence(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        if len(s) == 0:
            return True
        s_idx = 0
        
        for i in range(len(t)):
            if s[s_idx] == t[i]:
                s_idx += 1
                if s_idx == len(s):
                    return True
                
        return False
        
# @lc code=end

File: solutions_java/arrays/392.is-subsequence.java
/*
 * @lc app=leetcode id=392 lang=java
 *
 * [392] Is Subsequence
 */

// @lc code=start
class Solution {
    public boolean isSubsequence(String s, String t) {
        int sLen = s.length();
        int tLen = t.length();

        if (sLen < 1) return true;
        int sIdx = 0;
        int tIdx = 0;

        while (tIdx < tLen) {
          if (s.charAt(sIdx) == t.charAt(tIdx)) {
            sIdx++;
          }
          tIdx++;
          if (sIdx == sLen) return true;
        }
        return false;
    }
}
// @lc code=end

